File: RelatedWordsAPI/Models/Page.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace RelatedWordsAPI.Models
{
    public enum PageProcessingStatus
    {
        NotStarted,
        Processing,
        Filtered,
        Finished,
        Canceled,
        Failed,
        Unknown,
    }
    public class Page
    {
        public int? PageId { get; set; }
        public int? ProjectId { get; set; }
        public Project Project { get; set; }
        public string OriginalContent { get; set; }
        public string FilteredContent { get; set; }
        public string Url { get; set; }
        public PageProcessingStatus ProcessingStatus { get; set; }
        public ICollection<Sentence> Sentences { get; }
        public ICollection<WordPage> WordPage { get; }
    }

}

File: RelatedWordsUI/related-words-ui/src/models/Page.js
import isDefinedOrNull from './helpers/isDefinedOrNull';

class Page {
  constructor(props) {
    this.projectId = isDefinedOrNull(props.projectId);
    this.pageId = isDefinedOrNull(props.pageId);
    this.originalContent = isDefinedOrNull(props.originalContent);
    this.filteredContent = isDefinedOrNull(props.filteredContent);
    this.url = isDefinedOrNull(props.url);
    this.processingStatus = isDefinedOrNull(props.processingStatus);
  }

  processingStatusToString() {
    switch(this.processingStatus) {
      case 0:
        return 'Not Started';
      case 1:
        return 'Processing';
      case 2:
        return 'Filtered';
      case 3:
        return 'Finished';
      case 4:
        return 'Canceled';
      case 5:
        return 'Failed';
      case 6:
        return 'Unknown';
      default:
        return 'Unknown';
    }
  }
}

export default Page;